src/hmem.c

```diff
@@ -551,7 +551,7 @@ void ofi_hmem_set_iface_filter(const char* iface_filter_str, bool* filter)
 	/* always enable system hmem interface */
 	filter[FI_HMEM_SYSTEM] = true;
 
-	entry = strtok(iface_filter_str, token);
+	entry = strtok((char *)iface_filter_str, token);
```

Comment:
`strtok` modifies the input string. It's unsafe to simply cast away `const`. The input string comes from environment variables and does not want to be modified.  Suggest use `strdup` to create a copy to use here.

Reply:
Thanks, updated.